students/DennisLee/lesson02/series.py

```diff
@@ -0,0 +1,86 @@
+def fibonacci(n: int) -> int:
+    """
+    Print a list of fibonacci numbers up to a certain series count.
+
+    :n:  The desired index number of the fibonacci series
+         (starting with position 0).
+
+    :return:  The value of the fibonacci series at the specified index,
+              where a fibonacci number is always equal to the sum of the
+              previous two numbers in the fibonacci series, and the
+              1st two fibonacci numbers have values of **0** and **1**.
+    """
+    return sum_series(n)
+
+def lucas(n: int) -> int:
+    """
+    Print a list of lucas numbers up to a certain series count.
+
+    :n:  The desired index number of the lucas series
+         (starting with position 0).
+
+    :return:  The value of the lucas series at the specified index,
+              where a lucas number is always equal to the sum of the
+              previous two numbers in the lucas series, and the
+              1st two lucas numbers have values of **2** and **1**.
+    """
+    return sum_series(n, 2)
+
+def sum_series(index_num: int,
+               index_0_value: int = 0,
+               index_1_value: int = 1) -> int:
+    """
+    Retrieve a particular number within a series that contains the sum
+    of the series's previous two values.
+
+    :index_num:  The position number to retrieve (zero-based index).
+
+    :index_0_value:  The value of the first value within the series.
+                     Default is the fibonacci series's first value (0).
+
+    :index_1_value:  The value of the second value within the series.
+                     Default is the fibonacci series's second value (0).
+
+    :return: The value of the specified index number within the series.
+    """
+
+    if index_num < 0:
+        return -1  # error code - indexes can't be negative
+    elif index_num == 0:
+        return index_0_value
+    elif index_num == 1:
+        return index_1_value
+    else:
+        x, y = index_0_value, index_1_value
+        for i in range(index_num - 1):
+            x, y = y, x + y
+        return y    
```

Comment:
This is really slick solution. Note that there is an alternative recursive solution that could be adapted to work for `sum_series` as well since we know that `fibonacci(n) = fibonacci(n-2) + fibonacci(n-1)`

Reply:
I saw that equation mentioned in the assignment description. For some reason it didn't occur to me to try a recursive function, otherwise I would've written the function that way (even though I try to avoid recursion whenever I can... :) ).

Reply:
me too ;)